AnimationProfiles/TPP/tpp_AP.pddl

```diff
@@ -0,0 +1,145 @@
+(define (animation TPP-Propositional)
+
+; Defines the Animation profile for the TPP propositional problem
+; 18/08/2023
+; By Shu Tong Luo
+;
+; Note: Given the nature of the problem domain, multiple depots / markets 
+; are visualised as overlapping as there is no initial state predicate.
+ 
+ 
+; Specifies the goods is loaded onto a truck at a given level
+   (:predicate loaded
+                  :parameters (?g ?t ?l)
+                  :effect(
+                  (equal (?t y) (add (?l y) 50))
+                  (equal (?g y) (add (?t y) 20))
+                  (assign (?g x) (function align_middle (objects ?g ?t)))
+                  )
+   )
+
+; Specifies the goods is stored at a given level
+   (:predicate stored
+                  :parameters (?g ?l)
+                  :effect (
+                    (equal (?g x) (add (?l x) 50))
+                    (assign (?g x y) (function distribute_within_objects_vertical (objects ?g ?l) (settings (spacebtw 30))))
+                  )
+   )
+
+; Specifies that goods are available at the market at a given level
+   (:predicate on-sale
+                  :parameters (?g ?m ?l)
+                  :effect (
+                  (equal (?g x) (add (?m x) 20))
+                  )
+   )
```

Comment:
I coudn't observe any visual effect when on-sale levels of a particular good were reduced.  What's the proposed method to visualise the available number of items?

Reply:
I'd also aim to visualise the ready-to-load. Both on-sale and ready-to-load change when the buy action is executed.